src/providers/mssql/qgsmssqlnewconnection.cpp

```diff
@@ -252,6 +253,6 @@ void QgsMssqlNewConnection::showHelp()
 void QgsMssqlNewConnection::updateOkButtonState()
 {
   QListWidgetItem *item = listDatabase->currentItem();
-  bool enabled = !txtName->text().isEmpty() && !txtHost->text().isEmpty() && item;
+  bool enabled = !txtName->text().isEmpty() && !( txtService->text().isEmpty() && txtHost->text().isEmpty() ) && item;
```

Comment:
Minor thing, but I think this would be more readable if we flipped the logic:

```suggestion
  bool disabled = txtName->text().isEmpty() || ( txtService->text().isEmpty() && txtHost->text().isEmpty() ) || !item;
```

Reply:
Yes, indeed. Thanks for the suggestion. I've force pushed my branch with a new commit.